server/odc-service/src/main/java/com/oceanbase/odc/service/resource/k8s/model/K8sDeployment.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.resource.k8s.model;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.oceanbase.odc.service.resource.ResourceID;
+import com.oceanbase.odc.service.resource.ResourceLocation;
+import com.oceanbase.odc.service.resource.ResourceOperator;
+import com.oceanbase.odc.service.resource.ResourceState;
+
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+
+/**
+ * {@link K8sDeployment}
+ *
+ * @author yh263208
+ * @date 2024-09-10 11:18
+ * @since ODC_release_4.3.2
+ */
+@Getter
+@Setter
+@NoArgsConstructor
+public class K8sDeployment extends V1Deployment implements K8sResource {
+
+    public static final String TYPE = "K8S_DEPLOYMENT";
+    private ResourceState resourceState;
+    private ResourceLocation resourceLocation;
+    private ResourceOperator<K8sPod, K8sPod> resourceOperator;
```

Comment:
why in K8sDeployment  contains a ResourceOperator variables. if it's used for getK8sPods function. may remove this function out and implement as a tool function it better

Reply:
I have add this operation into the constructor